assets/js/modules/analytics-4/datastore/custom-dimensions.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * `modules/analytics-4` data store: custom-dimensions store.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+import { isPlainObject } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
+import { createReducer } from '../../../googlesitekit/data/create-reducer';
+import { isValidPropertyID } from '../utils/validation';
+
+const customDimensionFields = [
+	'parameterName',
+	'displayName',
+	'description',
+	'scope',
+	'disallowAdsPersonalization',
+];
+
+const fetchCreateCustomDimensionStore = createFetchStore( {
+	baseName: 'createCustomDimension',
+	controlCallback: ( { propertyID, customDimension } ) =>
+		API.set( 'modules', 'analytics-4', 'create-custom-dimension', {
+			propertyID,
+			customDimension: {
+				...customDimension,
+				scope: 'EVENT',
+			},
```

Comment:
This also isn't required by the AC, but also, while all of the custom dimensions we need now will be event scoped, we shouldn't limit the action to only creating such dimensions. Later if we did have such a requirement, it would be better to fail the request rather than silently create it as event-scoped instead.

Reply:
Thanks for the feedback. The decision to hardcode the `scope` as `EVENT` was made following a discussion with @nfmohit during the IB phase. However, in light of your feedback and considering future flexibility:

1. The backend logic has been implemented to handle cases where the `scope` might not be set. In such instances, it defaults to `EVENT`.
2. If a `scope` value is provided, it ensures that it's validated against the [DimensionScope](https://developers.google.com/analytics/devguides/config/admin/v1/rest/v1beta/properties.customDimensions#DimensionScope) enum values.

Reply:
Updated ✅